app/src/androidTest/java/ch/hikemate/app/ui/guide/GuideScreenTest.kt

```diff
@@ -0,0 +1,108 @@
+
+  @Mock private lateinit var mockNavigationActions: NavigationActions
+
+  private val testAppTopics =
+      listOf(
+          GuideTopic(
+              titleResId = R.string.guide_section_navigation,
+              contentResId = R.string.guide_content_navigation,
+              actionRoute = Route.MAP))
+
+  private val testHikingTopics =
+      listOf(
+          GuideTopic(
+              titleResId = R.string.guide_section_weather,
+              contentResId = R.string.guide_content_weather))
+
+  @Before
+  fun setup() {
+    mockNavigationActions = mock(NavigationActions::class.java)
+    setupGuideScreen()
+  }
+
+  @Test
+  fun guideScreen_displayHeader() {
+    composeTestRule.onNodeWithTag(GuideScreen.GUIDE_HEADER).assertExists()
+  }
+
+  @Test
+  fun guideScreen_displaysAppTopics() {
+    composeTestRule
+        .onNodeWithTag("${GuideScreen.TOPIC_CARD}_${testAppTopics[0].titleResId}")
+        .assertExists()
+  }
+
+  @Test
+  fun guideScreen_displaysHikingTopics() {
+    composeTestRule
+        .onNodeWithTag("${GuideScreen.TOPIC_CARD}_${testHikingTopics[0].titleResId}")
+        .assertExists()
+  }
+
+  @Test
+  fun topicCard_expandsWhenClicked() {
+    val topicCard = "${GuideScreen.TOPIC_CARD}_${testAppTopics[0].titleResId}"
+    val topicContent = "${GuideScreen.TOPIC_CONTENT}_${testAppTopics[0].titleResId}"
+
+    composeTestRule
+        .onNodeWithTag(topicCard)
+        .onChildren()[0] // Access the clickable Column inside Card
+        .performClick()
+
+    // The parameter useUnmergedTree is necessary because of the animation.
+    composeTestRule.onNodeWithTag(topicContent, useUnmergedTree = true).assertExists()
+  }
+
+  @Test
+  fun topicHeader_containsCorrectElements() {
+    val topicCard = "${GuideScreen.TOPIC_HEADER}_${testAppTopics[0].titleResId}"
+
+    composeTestRule.onNodeWithTag(topicCard).assertExists().assertIsDisplayed()
+  }
+
+  @Test
+  fun expandableContent_animatesCorrectly() {
+    val topicCard = "${GuideScreen.TOPIC_CARD}_${testAppTopics[0].titleResId}"
+    val topicContent = "${GuideScreen.TOPIC_CONTENT}_${testAppTopics[0].titleResId}"
+
+    // Initial state - content should not exist
+    composeTestRule.onNodeWithTag(topicContent, useUnmergedTree = true).assertDoesNotExist()
+
+    // Expand card
+    composeTestRule.onNodeWithTag(topicCard).onChildren()[0].performClick()
+
+    // After animation, content should exist
+    composeTestRule.waitForIdle()
+    composeTestRule.onNodeWithTag(topicContent, useUnmergedTree = true).assertExists()
+  }
+
+  @Test
+  fun guideScreen_hasCorrectPadding() {
+    composeTestRule.onNodeWithTag(GuideScreen.GUIDE_SCREEN).assertExists().assertHasNoClickAction()
+  }
+
+  private fun setupGuideScreen() {
+    composeTestRule.setContent {
+      GuideScreen(
+          navigationActions = mockNavigationActions,
+          appTopics = testAppTopics,
+          hikingTopics = testHikingTopics)
+    }
+  }
+}
```

Comment:
Warning: You currently don't test the Guide.kt class. I understand thats it's a bit difficult to come up with sensible tests, but I think you should add a few, like a test that checks if the Guides are initialized properly (initializing a guide and checking if the values match), a test that confirms that your two lists of guides are both not emtpy, and a test that asserts that a guide with no button action has a `null` value for actionRoute.

Reply:
Fixed.